src/main/java/tc/oc/pgm/modules/ForfeitMatchModule.java

```diff
@@ -0,0 +1,148 @@
+package tc.oc.pgm.modules;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+import net.md_5.bungee.api.ChatColor;
+import net.md_5.bungee.api.chat.ClickEvent;
+import net.md_5.bungee.api.chat.HoverEvent;
+import org.bukkit.command.CommandSender;
+import tc.oc.component.Component;
+import tc.oc.component.types.PersonalizedText;
+import tc.oc.component.types.PersonalizedTranslatable;
+import tc.oc.named.NameStyle;
+import tc.oc.pgm.Config;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.match.MatchModule;
+import tc.oc.pgm.match.MatchModuleFactory;
+import tc.oc.pgm.module.ModuleLoadException;
+import tc.oc.pgm.teams.TeamMatchModule;
+import tc.oc.pgm.timelimit.TimeLimitMatchModule;
+
+public class ForfeitMatchModule extends MatchModule {
+
+  public static class Factory implements MatchModuleFactory<ForfeitMatchModule> {
+    @Override
+    public ForfeitMatchModule createMatchModule(Match match) throws ModuleLoadException {
+      return new ForfeitMatchModule(match);
+    }
+  }
+
+  private boolean enabled;
+  private boolean broadcastEnabled;
+  private int maxPlayers;
+
+  public ForfeitMatchModule(Match match) {
+    super(match);
+    enabled = Config.Forfeit.enabled();
+    broadcastEnabled = Config.Forfeit.isBroadcastEnabled();
+    maxPlayers = Config.Forfeit.getMaxPlayerCount();
+  }
+
+  public boolean isEnabled() {
+    return enabled;
+  }
+
+  public boolean setEnabled(boolean on) {
+    enabled = on;
+    return isEnabled();
+  }
+
+  public int getTotalForfeits() {
+    return match.getParticipants().stream()
+        .filter(MatchPlayer::hasForfeit)
+        .collect(Collectors.counting())
+        .intValue();
+  }
+
+  public int getMatchPlayerCount() {
+    return match.getParticipants().size();
+  }
+
+  public void forfeit(MatchPlayer player, CommandSender sender) {
+    if (isForfietPossible()) {
+      player.setForfeit(!player.hasForfeit());
+      broadcastForfeit(player, player.hasForfeit());
+      calcForfeit(player);
+    } else {
+      player.sendWarning(new PersonalizedTranslatable("command.forfeit.notPossible"));
+    }
+  }
+
+  /*
+   * Current Restrictions:
+   * 1. If the match has a time limit.
+   * 2. If the match does not contain teams.
+   * 3. If player count is more than the max-player config.
+   */
+  private boolean isForfietPossible() {
+    Optional<TeamMatchModule> tmm = match.getModule(TeamMatchModule.class);
+    Optional<TimeLimitMatchModule> tlmm = match.getModule(TimeLimitMatchModule.class);
+
+    boolean hasTeams = tmm.isPresent();
+    boolean hasTimeLimit = tlmm.isPresent() && tlmm.get().hasTimeLimit();
+    boolean underPlayerLimit = maxPlayers != 0 ? getMatchPlayerCount() <= maxPlayers : true;
```

Comment:
Hard player count limit isn't the best way to go. Can we change `maxPlayers` to a percentage of players instead?